ribbon-eureka/src/main/java/com/netflix/niws/loadbalancer/DiscoveryEnabledNIWSServerList.java

```diff
@@ -198,6 +198,12 @@ public void initWithNiwsConfig(IClientConfig clientConfig) {
         return serverList;
     }
 
+    protected DiscoveryEnabledServer createServer(final InstanceInfo instanceInfo, boolean useSecurePort, boolean useIpAddr) {
+        DiscoveryEnabledServer server = new DiscoveryEnabledServer(instanceInfo, useSecurePort, useIpAddr);
+        server.setZone(DiscoveryClient.getZone(instanceInfo));
```

Comment:
We should improve the API, but for now, can you change from the deprecated `DiscoveryClient.getZone` to the following?
First, get hold of EurekaClient (there is already an instance), then
```
EurekaClientConfig clientConfig = eurekaClient.getEurekaClientConfig();
String[] availZones = clientConfig.getAvailabilityZones(clientConfig.getRegion());
String instanceZone = InstanceInfo.getZone(availZones, myInstanceInfo);
server.setZone(instanceZone);
```

Reply:
Absolutely. Although it looks like getEurekaClientConfig() only exists on DiscoveryClient, not EurekaClient - so is it ok for me to cast to DiscoveryClient if it is an instanceof?

Reply:
Hmm, looks like ribbon still depends on a really ancient version of eureka, that method was added to the interface from 1.6.0 onwards. Do you mind also updating the eureka dependency to 1.7.2? (1.7.2 is the current latest Java7 compatible version). Thanks.